lib/segment/src/index/field_index/map_index/mod.rs

```diff
@@ -362,25 +371,31 @@ impl PayloadFieldIndex for MapIndex<SmolStr> {
                     .push(PrimaryCondition::Condition(condition.clone()));
                 Ok(estimation)
             }
-            Some(Match::Any(MatchAny {
-                any: AnyVariants::Keywords(keywords),
-            })) => {
-                let estimations = keywords
-                    .iter()
-                    .map(|keyword| {
-                        let keyword_condition = FieldCondition::new_match(
-                            condition.key.clone(),
-                            keyword.to_owned().into(),
-                        );
-                        self.match_cardinality(keyword.as_str())
-                            .with_primary_clause(PrimaryCondition::Condition(keyword_condition))
-                    })
-                    .collect::<Vec<_>>();
-                Ok(combine_should_estimations(
-                    &estimations,
-                    self.get_indexed_points(),
-                ))
-            }
+            Some(Match::Any(MatchAny { any: any_variant })) => match any_variant {
+                AnyVariants::Keywords(keywords) => {
+                    let estimations = keywords
+                        .iter()
+                        .map(|keyword| self.match_cardinality(keyword.as_str()))
+                        .collect::<Vec<_>>();
+                    let estimation = if estimations.is_empty() {
+                        CardinalityEstimation::exact(0)
+                    } else {
+                        combine_should_estimations(&estimations, self.get_indexed_points())
+                    };
```

Comment:
Can we move the `is_empty` check into `combine_should_estimations` as early return?

That would remove branching here and could potentially solve similar cases in the future.

Reply:
no, cause `should` will condition will match all points if it is empty, while `match: any` will not

Reply:
Makes total sense. Thanks for clarifying!